src/commands/plugin-search.php

```diff
@@ -2,116 +2,206 @@
-		->addArgument( 'plugin-slug', InputArgument::REQUIRED, "The slug of the plugin to search for. This is an exact match against the plugin installation folder name,\nthe main plugin file name without the .php extension, and the Text Domain.\n" )
-		->addOption( 'partial', null, InputOption::VALUE_OPTIONAL, "Optional.\nUse for general text/partial match search. Using this option will also search the plugin Name field." );
-	}
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function configure(): void {
+		$this->setDescription( 'Search all Team51 WPCOM Jetpack sites for a specific plugin.' )
+			->setHelp( "This command will output a list of Jetpack sites connected to the a8cteam51 account where a particular plugin is installed.\nThe search can be made for an exact match plugin slug, or\na general text search. Letter case is ignored in both search types.\nExample usage:\nplugin-search woocommerce\nplugin-search woo --partial\n" );
 
-	protected function execute( InputInterface $input, OutputInterface $output ) {
-		$plugin_slug = strtolower( $input->getArgument( 'plugin-slug' ) );
+		$this->addArgument( 'plugin-slug', InputArgument::REQUIRED, "The slug of the plugin to search for. This is an exact match against the plugin installation folder name,\nthe main plugin file name without the .php extension, and the Text Domain.\n" )
+			->addOption( 'partial', null, InputOption::VALUE_NONE, "Optional.\nUse for general text/partial match search. Using this option will also search the plugin Name field." );
+	}
 
-		$api_helper = new API_Helper;
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function initialize( InputInterface $input, OutputInterface $output ): void {
+		maybe_define_console_verbosity( $output->getVerbosity() );
 
-		$output->writeln( '<info>Fetching list of sites...<info>' );
+		$this->plugin_slug = \strtolower( $input->getArgument( 'plugin-slug' ) );
+		$this->partial     = (bool) $input->getOption( 'partial' );
+	}
 
-		$sites = $api_helper->call_wpcom_api( 'rest/v1.1/jetpack-blogs/', array() );
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ): int {
+		$partial_match_text = $this->partial ? 'partial' : 'exact';
+		$output->writeln( "<fg=magenta;options=bold>Searching through the WPCOM Jetpack sites for the plugin with the slug `$this->plugin_slug` ($partial_match_text match).</>" );
 
+		// Get the list of sites.
+		$sites = get_wpcom_jetpack_sites();
 		if ( empty( $sites ) ) {
 			$output->writeln( '<error>Failed to fetch sites.<error>' );
-			exit;
-		}
-
-		$site_list = array();
-		foreach ( $sites->blogs->blogs as $site ) {
-			$site_list[] = array( $site->userblog_id, $site->domain );
+			return 1;
```

Comment:
Because Symfony's `Symfony\Component\Console\Command\Command` expects an int value as a return ([see here](https://github.com/symfony/console/blob/3b03de0c529b6efb02deb55a13a9059d40794373/Command/Command.php#L214)).

A return value of 0 means everything went well, and anything non-zero means there was an error. If we call `exit` directly we have the same result on the terminal, i.e. the program ends, but it doesn't really tell us why. If we return different ints we can assign meaning to them, e.g. a return value of 5 could indicate a lack of adequate permissions. Since the logic is not too complicated, I return 0 on success and 1 on failure. This is especially helpful if we pipe the command through other applications which will then expect a proper return code. 

At the same time, if we call a command from inside a different command in the CLI tool (we already do that in some places), a call to `exit` inside that "child" command will close the whole application. If we return an int instead of calling exit, we can do proper error handling in the "parent" command.

But ultimately I am doing it like that because it's what Symfony expects us to do (see first paragraph above).